test/plugin/example_plugin_test.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//
+const testIODeadline = 2 * time.Second
+
+// readNLines reads n lines from net.Conn using testIODeadline as the read
+// deadline to avoid tests that hang forever.
+// NOTE: lines are 1-indexed.
+func readNLines(conn net.Conn, n int) ([]string, error) {
+	defer func() {
+		_ = conn.SetReadDeadline(time.Time{})
+	}()
+
+	var lines []string
+	scanner := bufio.NewScanner(conn)
+
+	i := 0
+	_ = conn.SetReadDeadline(time.Now().Add(testIODeadline))
+	for i < n && scanner.Scan() {
+		i++
+		lines = append(lines, scanner.Text())
+
+		_ = conn.SetReadDeadline(time.Now().Add(testIODeadline))
```

Comment:
nit: I know that we're explicitly ignoring the error but I think it's really superfluous for `SetReadDeadline`/`SetWriteDeadline` in this file